cmd/neofs-node/object.go

```diff
@@ -665,19 +681,73 @@ func (o objectSource) Head(ctx context.Context, addr oid.Address) (*objectSDK.Ob
 }
 
 func (o objectSource) Search(ctx context.Context, cnr cid.ID, filters objectSDK.SearchFilters) ([]oid.ID, error) {
```

Comment:
This API can be reworked to be sv2-compatible as well. It is important to have it being used properly. For example, tombstone verifier for v2 doesn't need all oids, it can set lower limit. For v1 it can avoid subsequent link object HEAD by requesting payload size (see #3324, it's not as reliable as children check, but it's good enough for old v1 objects).